zapisy/apps/theses/views.py

```diff
@@ -150,16 +145,13 @@ def view_thesis(request, id):
 @login_required
 def gen_form(request, id, studentid):
     """Display form to print for specific student assigned to a thesis."""
-    thesis = get_object_or_404(Thesis, id=id)
+    thesis = get_object_or_404(Thesis.objects.visible(request.user), id=id)
     try:
         first_student = thesis.students.get(id=studentid)
     except Student.DoesNotExist:
         raise Http404("No Student matches the given query.")
 
-    user_privileged_for_thesis = thesis.is_among_advisors(
-        request.user) or request.user.is_staff or is_theses_board_member(request.user)
-    user_allowed_to_generate = user_privileged_for_thesis or (
-        thesis.has_been_accepted and thesis.is_student_assigned(request.user))
+    user_allowed_to_generate = thesis.is_user_allowed_to_generate_declarations(request.user)
```

Comment:
```suggestion
    if not thesis.is_user_allowed_to_generate_declarations(request.user):
```
z usunięciem kolejnego wiersza.